packages/spacecat-shared-ims-client/src/clients/ims-client.js

```diff
@@ -297,4 +302,46 @@ export default class ImsClient {
       admins,
     };
   }
+
+  /**
+   * Fetch a subset of properties from a user's IMS profile, given their access token.
+   * @param {string} imsAccessToken A valid IMS user access token
+   * @returns {Promise<{userId, email, organizations: string[]}>} Fields from the user's profile
+   */
+  async getImsUserProfile(imsAccessToken) {
+    // Helper to pull the unique organization ID values from an array of role entries
+    function getOrganizationList(roles) {
+      return [...new Set(roles.map((roleEntry) => roleEntry.organization))];
+    }
+
+    try {
+      const startTime = process.hrtime.bigint();
+
+      const profileResponse = await httpFetch(
+        `https://${this.config.imsHost}${IMS_PROFILE_ENDPOINT}`,
+        {
+          headers: {
+            Authorization: `Bearer ${imsAccessToken}`,
+          },
+        },
+      );
+
+      if (!profileResponse.ok) {
+        throw new Error(`IMS getImsUserProfile request failed with status: ${profileResponse.status}`);
+      }
+
+      const profile = await profileResponse.json();
```

Comment:
you could use destructuring:
```suggestion
const { userId, email, roles } = await profileResponse.json();
```